tests/test_misc.py

```diff
@@ -17,3 +23,45 @@ def test_pyc_f_3(self):
                  'sp/foo/__pycache__/utils.cpython-34.pyc'),
         ]:
             self.assertEqual(_link.pyc_f(f, (3, 4, 2)), r)
+
+
+def test_pathtype():
+    hardlink = PathType("hardlink")
+    assert str(hardlink) == "hardlink"
+    assert hardlink.__json__() == 'hardlink'
+
+    softlink = PathType("softlink")
+    assert str(softlink) == "softlink"
+    assert softlink.__json__() == "softlink"
+
+
+def test_entity_encoder(tmpdir):
+    test_file = join(str(tmpdir), "test-file")
+    test_json = {"a": PathType("hardlink"), "b": 1}
+    with open(test_file, "w") as f:
+        json.dump(test_json, f, cls=EntityEncoder)
+
+    with open(test_file, "r") as f:
+        json_file = json.load(f)
+    assert json_file == {"a": "hardlink", "b": 1}
+
+
+@pytest.mark.skipif(on_win, reason="link not available on win/py2.7")
+def test_crossplatform_st_link(tmpdir):
+    test_file = join(str(tmpdir), "test-file")
+    test_file_linked = join(str(tmpdir), "test-file-linked")
+    test_file_link = join(str(tmpdir), "test-file-link")
+
+    open(test_file, "a").close()
+    open(test_file_link, "a").close()
+    link(test_file_link, test_file_linked)
+    assert 1 == CrossPlatformStLink.st_nlink(test_file)
+    assert 2 == CrossPlatformStLink.st_nlink(test_file_link)
+    assert 2 == CrossPlatformStLink.st_nlink(test_file_linked)
+
+
+def test_crossplatform_st_link_on_win(tmpdir):
```

Comment:
Might want a platform skip on things other than windows.  Just invert the logic from your test one up.